identity_vc/src/credential.rs

```diff
@@ -0,0 +1,62 @@
+use anyhow::Result;
+use chrono::DateTime;
+
+use crate::common::{Context, Issuer, Object, OneOrMany, URI};
+
+/// A `Credential` represents a set of claims describing an entity.
+///
+/// `Credential`s can be combined with `Proof`s to create `VerifiableCredential`s.
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Credential {
```

Comment:
Several fields here are represented as `Object`. Most of them should probably be structs as they do have a bit of a pre-defined structure.

An example of how `Evidence` could be represented.

```rust
#[derive(Debug, Deserialize, Serialize)]
pub struct Evidence {
  id: Option<String>,
  #[serde(rename = "type")]
  types_: String,
  #[serde(flatten)]
  object: Object,
}
```

Since many fields are optional and no data is being discarded, I don't think it's too high of a priority.

@tensor-programming thoughts?

Reply:
Its up to you whether or not you want to include them.  I see you are basically nesting hashmaps for the object representation which is fine.  The one thing I would worry about is allowing convoluted json structures to be valid since this doesn't make any assumptions about the types of keys you are accepting. Aside from that, it should be acceptable.